omnibus/config/projects/agent.rb

```diff
@@ -197,85 +214,79 @@
 # Dependencies
 # ------------------------------------
 
-# Datadog agent
-dependency 'datadog-agent'
+if do_build
+  # Datadog agent
+  dependency 'datadog-agent'
 
-# System-probe
-if linux_target? && !heroku_target?
-  dependency 'system-probe'
-end
+  # System-probe
+  if linux_target? && !heroku_target?
+    dependency 'system-probe'
+  end
 
-if osx_target?
-  dependency 'datadog-agent-mac-app'
-end
+  if osx_target?
+    dependency 'datadog-agent-mac-app'
+  end
 
-if with_python_runtime? "2"
-  dependency 'pylint2'
-  dependency 'datadog-agent-integrations-py2'
-end
+  if with_python_runtime? "2"
+    dependency 'pylint2'
+    dependency 'datadog-agent-integrations-py2'
+  end
 
-if with_python_runtime? "3"
-  dependency 'datadog-agent-integrations-py3'
-end
+  if with_python_runtime? "3"
+    dependency 'datadog-agent-integrations-py3'
+  end
 
-if linux_target?
-  dependency 'datadog-security-agent-policies'
-end
+  if linux_target?
+    dependency 'datadog-security-agent-policies'
+  end
 
-# Include traps db file in snmp.d/traps_db/
-dependency 'snmp-traps'
+  # Include traps db file in snmp.d/traps_db/
+  dependency 'snmp-traps'
 
-# Additional software
-if windows_target?
-  if ENV['WINDOWS_DDNPM_DRIVER'] and not ENV['WINDOWS_DDNPM_DRIVER'].empty?
-    dependency 'datadog-windows-filter-driver'
-  end
-  if ENV['WINDOWS_APMINJECT_MODULE'] and not ENV['WINDOWS_APMINJECT_MODULE'].empty?
-    dependency 'datadog-windows-apminject'
+  # Additional software
+  if windows_target?
+    if ENV['WINDOWS_DDNPM_DRIVER'] and not ENV['WINDOWS_DDNPM_DRIVER'].empty?
+      dependency 'datadog-windows-filter-driver'
+    end
+    if ENV['WINDOWS_APMINJECT_MODULE'] and not ENV['WINDOWS_APMINJECT_MODULE'].empty?
+      dependency 'datadog-windows-apminject'
+    end
+    if ENV['WINDOWS_DDPROCMON_DRIVER'] and not ENV['WINDOWS_DDPROCMON_DRIVER'].empty?
+      dependency 'datadog-windows-procmon-driver'
+      ## this is a duplicate of the above dependency in linux
+      dependency 'datadog-security-agent-policies'
+    end
   end
-  if ENV['WINDOWS_DDPROCMON_DRIVER'] and not ENV['WINDOWS_DDPROCMON_DRIVER'].empty?
-    dependency 'datadog-windows-procmon-driver'
-    ## this is a duplicate of the above dependency in linux
-    dependency 'datadog-security-agent-policies'
+
+  # this dependency puts few files out of the omnibus install dir and move them
+  # in the final destination. This way such files will be listed in the packages
+  # manifest and owned by the package manager. This is the only point in the build
+  # process where we operate outside the omnibus install dir, thus the need of
+  # the `extra_package_file` directive.
+  # This must be the last dependency in the project.
+  dependency 'datadog-agent-finalize'
+  dependency 'datadog-cf-finalize'
+  if do_package
+    dependency "init-scripts-agent"
```

Comment:
This looks redundant with the conditional right after this (line 273).

Reply:
Ah indeed. It's not entirely redundant as we need the one on line 273 for heroku (which for now still does build & packaging in a single step) but the flow is incorrect as we shouldn't care about packaging if we were building.

I'll push a fix shortly